travel/src/main/java/com/tx/travel/service/AuthService.java

```diff
@@ -25,55 +24,47 @@ public AuthService(final UserRepository userRepository, final RoleRepository rol
   }
 
   public void findByUsernameOrEmail(@NotNull final String username, @NotNull final String email)
-      throws UsernameAlreadyExistsException {
+      throws UsernameAlreadyExistsException,EmailAlreadyExistsException {
     final Optional<User> userByUsername = userRepository.findByUsername(username);
     if (userByUsername.isPresent()) {
       throw new UsernameAlreadyExistsException(username);
     }
     final Optional<User> userByEmail = userRepository.findByEmail(email);
     if (userByEmail.isPresent()) {
-      throw new UsernameAlreadyExistsException(email);
+      throw new EmailAlreadyExistsException(email);
```

Comment:
method is called `findByUsernameOrEmail` and we are returning exception `EmailAlreadyExistsException` should not work like this?